### PR TITLE
Making Location Mandatory in New-AzPublicIpPrefix and New-AzPublicIpAddress

### PR DESCRIPTION
**Pull request by allegradomel**
## Description
Per https://github.com/Azure/azure-powershell/issues/10953, made Location a mandatory parameter in New-AzPublicIpPrefix and New-AzPublicIpAddress since this parameter is required to successfully run the cmdlet. Made Name mandatory in New-AzPublicIpAddress for the same reason. 

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Network/Network/help/New-AzPublicIpAddress.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ Creates a public IP address.
 ## SYNTAX
 
 ```
-New-AzPublicIpAddress [-Name <String>] -ResourceGroupName <String> [-Location <String>] [-Sku <String>]
+New-AzPublicIpAddress -Name <String> -ResourceGroupName <String> -Location <String> [-Sku <String>]
```
Comment (markcowl) [inline src/Network/Network/help/New-AzPublicIpAddress.md]: This change cannot be kade in the help file.  It must be made in cmdlet code.  The syntax portion of help is generated, so this change will be wiped out the next time help is generated.  Additionally, PowerShell users will only gain the benefits of the parameter being mandatory (like being prompted if the parameter is not provided)

Comment (allegradomel) [reply src/Network/Network/help/New-AzPublicIpAddress.md]: The parameter is already mandatory in the sense that the user receives an error if they don't supply it. However, the help file/documentation doesn't specify it is required. Under **Parameters** in the help file, I changed **Required** to be **True** for these. Is this incorrect? 

Thanks!

**Comment by allegradomel** [comment]
> https://github.com/Azure/azure-powershell/blob/master/src/Network/Network/PublicIpAddress/NewAzurePublicIpAddressCommand.cs#L50
> https://github.com/Azure/azure-powershell/blob/master/src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs#L49
> As you can see in these two lines of code, location is still optional.

Thanks for pointing it out, new to Powershell code so I didn't realize :) Will make the necessary changes

**Comment by allegradomel** [comment]
> https://github.com/Azure/azure-powershell/blob/master/src/Network/Network/PublicIpAddress/NewAzurePublicIpAddressCommand.cs#L50
> https://github.com/Azure/azure-powershell/blob/master/src/Network/Network/PublicIpPrefix/NewAzurePublicIpPrefixCommand.cs#L49
> As you can see in these two lines of code, location is still optional.
 Right - will make these fixes

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by isra-fel** [comment]
Hi @allegradomel we are good with your change. But it's recognized as a breaking change by static analysis tool. Please follow this guide to suppress the CI error. Thanks

https://github.com/Azure/azure-powershell/blob/master/documentation/Debugging-StaticAnalysis-Errors.md#breaking-changes

**Comment by allegradomel** [comment]
> Hi @allegradomel we are good with your change. But it's recognized as a breaking change by static analysis tool. Please follow this guide to suppress the CI error. Thanks
> https://github.com/Azure/azure-powershell/blob/master/documentation/Debugging-StaticAnalysis-Errors.md#breaking-changes

Great will do

**Comment by msJinLei** [comment]
@allegradomel Please resolve the conflicts.

**Comment by isra-fel** [comment]
We are closing the PR @allegradomel
If you decide to continue working on the PR, we suggest create a new branch and cherry-pick your commits.
(10000 commits shows up in the history because we force-pushed to master a few months ago due to a security issue. Sorry for the inconvenience!)

**Pull request closed without merging**